bson/_cbsonmodule.c

```diff
@@ -1760,6 +1763,9 @@ int write_dict(PyObject* self, buffer_t buffer,
                             if (msg) {
                                 // Prepend doc to the existing message
                                 PyObject *dict_str = PyObject_Str(dict);
+                                if (dict_str ==  NULL) {
+                                    return 0;
+                                }
                                 PyObject *new_msg = PyUnicode_FromFormat("Invalid document %s | %s", PyUnicode_AsUTF8(dict_str), PyUnicode_AsUTF8(msg));
```

Comment:
Could you fix the indentation on line 1776?

Reply:
Also I believe PyUnicode_AsUTF8 can return NULL on error too. We'll need checks there too. Unless there's a simpler API to construct this string that does the NULL checks for us.

Reply:
Fixed